### PR TITLE
BAQE-338: Increase coverage

### PR DESCRIPTION
**Pull request by jomarko**
This PR increases coverages of:
- Guided Rule Template backend
  - Validation of guided rule template
  - Loading of enums
- Guided Rule Widgets - especially parts related to enums

@manstis please have a look, as this contains also some refactoring, I will not cherry pick into 7.5.x.

### CODE REVIEW

**Comment by jomarko** [comment]
jenkins please retest this

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java [hunk 1]
```diff
@@ -229,33 +230,16 @@ private void isLHSFieldTypeEquivalent(final String boundVariable,
...
-        if (values.length != this.dropDownData.getFixedList().length) {
-            return false;
-        }
-        for (int i = 0; i < values.length; i++) {
-            if (!values[i].equals(this.dropDownData.getFixedList()[i])) {
-                return false;
-            }
-        }
-        return true;
+    public static boolean isEnumEquivalent(final String[] values, final DropDownData dropDownData) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java]: :+1:

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/test/java/org/drools/workbench/screens/guided/rule/client/util/RefreshUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.runner.RunWith;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@WithClassesToStub(DateTimeFormat.class)
+@RunWith(GwtMockitoTestRunner.class)
+public class RefreshUtilTest {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/test/java/org/drools/workbench/screens/guided/rule/client/util/RefreshUtilTest.java]: Is it worth doing a test where both drop-downs refresh?

These are used by "dependent" enums, e.g. Country/Region drop-downs where Region would refresh if Country changed. A more complex example would be Country/Region/Town. Changes to Country should refresh Region which then changes and should refresh Towns. A change to Region would not refresh Country. 

If you feel these tests are sufficient then ignore me (some tests are better than none!)

File: drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-backend/src/test/java/org/drools/workbench/screens/guided/template/server/GuidedRuleTemplateEditorServiceImplCDITest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        super.setUp();
+        doReturn(syncBeanDef).when(syncBeanManager).lookupBean(any());
+        doReturn(asyncOracle).when(syncBeanDef).getInstance();
+
+        testedService = getReference(GuidedRuleTemplateEditorService.class);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.cleanup();
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-backend/src/test/java/org/drools/workbench/screens/guided/template/server/GuidedRuleTemplateEditorServiceImplCDITest.java]: Did you know the ```cleanup()``` method is failing sometimes (did you see @sterobin comment about one of the PRs for "Template Key Tooltip" failing - which on examination was a ```NPE``` thrown from ```cleanup()``` by the WELD container itself!)

Comment (jomarko) [reply drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-backend/src/test/java/org/drools/workbench/screens/guided/template/server/GuidedRuleTemplateEditorServiceImplCDITest.java]: I hope solved by https://github.com/kiegroup/drools-wb/pull/794

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/test/java/org/drools/workbench/screens/guided/rule/client/util/RefreshUtilTest.java (hunk 2 above)
Comment (jomarko) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/test/java/org/drools/workbench/screens/guided/rule/client/util/RefreshUtilTest.java]: @manstis I added some tests in this class, and also some formatting/imports cleanup, please could you re-review?

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/widget/DSLSentenceWidget.java [hunk 4]
```diff
@@ -250,9 +247,12 @@ private Widget getEnumDropdown(String variableDef,
...
         final boolean multipleSelect = isMultipleSelect(variableDef,
                                                         sentence);
-        DSLDropDown resultWidget = new DSLDropDown(variableDef,
+        DSLDropDown resultWidget = new DSLDropDown(modeller,
+                                                   variableDef,
+                                                   sentence,
                                                    value,
-                                                   multipleSelect);
+                                                   multipleSelect,
+                                                   dslDropDwon -> updateEnumDropDowns(dslDropDwon));
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/widget/DSLSentenceWidget.java]: Typo in parameter "dslDropD**wo**n"

File: drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/test/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    @Before
+    public void setUp() throws Exception {
+        testedFactory = spy(new TemplateDataCellFactory(oracle, dropDownManager, isReadOnly, eventBus));
+    }
+
+    @Test
+    public void testGetCell() throws Exception {
+        final String factType = "org.kiegroup.Car";
+        final String factField = "color";
+        final String dataType = DataType.DataTypes.NUMERIC.name();
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/test/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactoryTest.java]: Technically ```color``` should be a ```STRING```.

File: drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/test/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactoryTest.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        testedFactory.getCell(column);
+
+        verify(testedFactory, never()).makeSingleSelectionEnumCell(factType, factField, dataType);
+    }
+
+    @Test
+    public void testGetCellWhenEnumPresent() throws Exception {
+        final String factType = "org.kiegroup.Car";
+        final String factField = "color";
+        final String dataType = DataType.DataTypes.NUMERIC.name();
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/test/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactoryTest.java]: Technically ```color``` should be a ```STRING```.

**Comment by jomarko** [comment]
@manstis all should be resolved by the last commit.

**Comment by jomarko** [comment]
@manstis jenkins says green.

**Pull request merged**